Kinvey.Core/Query/KinveyQueryVisitor.cs

```diff
@@ -112,31 +112,65 @@ public override void VisitWhereClause(WhereClause whereClause, QueryModel queryM
 			{
 				BinaryExpression equality = whereClause.Predicate as BinaryExpression;
 				var member = equality.Left as MemberExpression;
+				var argument = equality.Right.ToString();
 
-				builderMongoQuery.Write ("\"" + mapPropertyToName[member.Member.Name] + "\"");
-				builderMongoQuery.Write (":");
-				builderMongoQuery.Write (equality.Right);
+				builderMongoQuery.Write("\"" + mapPropertyToName[member.Member.Name] + "\"");
+				builderMongoQuery.Write(":");
+
+				if (equality.Right.Type.Name.Equals("Boolean"))
```

Comment:
@vinaygahlawat is all the dereferencing in this block of code guaranteed to succeed? Is there any reason to be more defensive with nullable checks?